src/Symfony/Component/Routing/Annotation/Route.php

```diff
@@ -106,6 +106,22 @@ public function getName()
 
     public function setRequirements($requirements)
     {
+        if (isset($requirements['_method'])) {
+            if (0 === count($this->methods)) {
+                $this->methods = explode('|', $requirements['_method']);
```

Comment:
if you synchronize it this direction, then it must also be synchronized in the other direction (setting methods would need to set the _method requirement, as done in the Route class). But I don't see the point of doing it in the annotation class.